### PR TITLE
Added Heizou

### PR DESCRIPTION
**Pull request by Charlie-Zheng**
Added Heizou

### CODE REVIEW

File: internal/characters/heizou/attack.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// assuming it's 0.02s. Please verify
+var attackHitlagHaltFrame = [][]float64{{0.02}, {0.02}, {0.02}, {0, 0, 0.02}, {0.02}}
+var attackDefHalt = [][]bool{{true}, {true}, {true}, {false, false, true}, {true}}
+
+const normalHitNum = 5
+
+func init() {
+	attackFrames = make([][]int, normalHitNum)
+
+	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], attackHitmarks[0][0]+1)
```
Comment (shizukayuki) [inline internal/characters/heizou/attack.go:24]: 2nd arg should be longest anim/cancel, `21` for this case.

File: internal/characters/heizou/attack.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][0], attackHitmarks[2][0]+1)
+	attackFrames[2][action.ActionAttack] = 45
+	attackFrames[2][action.ActionCharge] = 46
+
+	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3][2], attackHitmarks[3][2]+1)
+	attackFrames[3][action.ActionAttack] = 36
+	attackFrames[3][action.ActionCharge] = 38
+
+	attackFrames[4] = frames.InitNormalCancelSlice(attackHitmarks[4][0], attackHitmarks[4][0]+1)
+	attackFrames[4][action.ActionAttack] = 66
```
Comment (shizukayuki) [inline internal/characters/heizou/attack.go:41]: Add `ActionCharge` with 500 frames or something as illegal action.

File: internal/characters/heizou/burst.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+func init() {
+	burstFrames = frames.InitAbilSlice(72)
+	burstFrames[action.ActionAttack] = 71
+	burstFrames[action.ActionSkill] = 71
+	burstFrames[action.ActionJump] = 70
+	burstFrames[action.ActionSwap] = 69
+
+}
+
+var hitmark = 34
```
Comment (shizukayuki) [inline internal/characters/heizou/burst.go:22]: Rename to `burstHitmark` and declare it as `const`.

File: internal/characters/heizou/burst.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	c.Core.QueueAttack(ai, combat.NewCircleHit(c.Core.Combat.PrimaryTarget(), 4, false, combat.TargettableEnemy), hitmark, hitmark)
+
+	//TODO: Check CD with or without delay, check energy consume frame
+	c.SetCD(action.ActionBurst, 12*60)
+	c.ConsumeEnergy(3)
+	return action.ActionInfo{
+		Frames:          frames.NewAbilFunc(burstFrames),
+		AnimationLength: burstFrames[action.InvalidAction],
+		CanQueueAfter:   burstFrames[action.ActionSwap],
+		State:           action.NormalAttackState,
```
Comment (shizukayuki) [inline internal/characters/heizou/burst.go:78]: Wrong state.

**Review by shizukayuki — CHANGES_REQUESTED** [review]
mostly just consistency stuff.

File: internal/characters/heizou/asc.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		}
+	}
+
+	c.Core.Events.Subscribe(event.OnDamage, swirlCB(), "heizou-a1")
+}
+
+func (c *char) a4() {
+
+	dur := 60 * 10
+	c.AddStatus("heizoua4", dur, true)
```
[the remark this answers is not in the document]
Comment (Charlie-Zheng) [reply internal/characters/heizou/asc.go:43]: It was in the original heizou implementation

[the remark this answers is not in the document]
Comment (shizukayuki) [reply internal/characters/heizou/asc.go:43]: It does nothing currently, unless there is configs that check for it.

**Review by shizukayuki — APPROVED** [review]

**Pull request merged**